src/main/java/org/apache/accumulo/testing/performance/tests/HerdingPT.java

```diff
@@ -0,0 +1,165 @@
+import java.util.Random;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import org.apache.accumulo.core.client.AccumuloClient;
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.BatchWriter;
+import org.apache.accumulo.core.client.MutationsRejectedException;
+import org.apache.accumulo.core.client.Scanner;
+import org.apache.accumulo.core.client.TableExistsException;
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.testing.performance.Environment;
+import org.apache.accumulo.testing.performance.PerformanceTest;
+import org.apache.accumulo.testing.performance.Report;
+import org.apache.accumulo.testing.performance.SystemConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HerdingPT implements PerformanceTest {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HerdingPT.class);
+
+  private static final byte[] COL_FAM = "pinky".getBytes();
+
+  private static final int NUM_ROWS = 1_000_000;
+  private static final int NUM_COLS = 10;
+  private static final int NUM_THREADS = 32;
+  private static final int NUM_LOAD_ATTEMPTS = 1_000;
+
+  private static final String TABLE_NAME = "herd";
+  private static final String DESCRIPTION = "Test herding performance with " + NUM_THREADS
+      + " threads attempting to " + "simultaneously load the same block " + NUM_LOAD_ATTEMPTS
+      + " times.";
+
+  private final Random random = new Random();
+
+  @Override
+  public SystemConfiguration getSystemConfig() {
+    Map<String,String> config = new HashMap<>();
+
+    config.put(Property.TABLE_BLOCKCACHE_ENABLED.getKey(), "true");
+
+    return new SystemConfiguration().setAccumuloConfig(config);
+  }
+
+  @Override
+  public Report runTest(final Environment env) throws Exception {
+    Report.Builder reportBuilder = Report.builder().id("herdingPT").description(DESCRIPTION)
+        .parameter("table_name", TABLE_NAME, "The name of the table.")
+        .parameter("num_threads", NUM_THREADS, "The number of threads.")
+        .parameter("num_load_attempts", NUM_LOAD_ATTEMPTS,
+            "The number of times the threads will attempt to load the same block.")
+        .parameter("num_rows", NUM_ROWS, "The number of rows that will be loaded into the table.");
+
+    final AccumuloClient client = env.getClient();
+    initTable(client);
+    long herdTime = getHerdingDuration(client);
+    reportBuilder.parameter("herd_time", herdTime, "The time (in ms) it took herding to complete.");
```

Comment:
`reportBuilder.parameter` should be replaced with `reportBuilder.result` for the resulting value of the performance test

Reply:
Code updated.